src/components/Workshops/index.tsx

```diff
@@ -102,31 +104,49 @@ export const Workshops = ({ events, tags, levels }: Props) => {
 
         return true;
       }),
-    [formattedWorkshops, currentLevels, currentTags, currentPriceFilters]
+    [formattedWorkshops, currentLevels, currentTags, currentPriceFilters],
   );
 
   const oldEvents = useMemo(
     () =>
       sortEvents(
         filteredEventsCalculated.filter(
-          (event) => event.normalizedDateStart < dateNow
+          (event) => event.normalizedDateStart < dateNow,
         ),
         sortType,
-        true
+        true,
       ),
-    [filteredEventsCalculated, sortType]
+    [filteredEventsCalculated, sortType],
   );
 
+  const [postsToShow, setPostsToShow] = useState([]);
+  const [next, setNext] = useState(5);
+
+  const loopWithSlice = (start, end) => {
+    const slicedPosts = oldEvents.slice(start, end);
+    arrayForHoldingPosts = [...arrayForHoldingPosts, ...slicedPosts];
```

Comment:
po co to trzymasz? 
`
const [numPostsToShow, setNumPostsToShow] = useState(5); 
const postsToShow = useMemo(() => {
    return oldEvents.slice(0, numPostsToShow);
}, [numPostsToShow, oldEvents]

const onBtnClick = () => setNumPostsToShow(numPostsToShow + POSTS_PER_PAGE)
`